### PR TITLE
Added "Why must all subprojects be inside a single directory?" to Subprojects page

### LINKED ISSUE #2766: Why all subproject should be inside a directory?
Hi guys,

I'm starting to build more complex build using meson and I'm loving it. I'm sure this is one of, if not the best, build system out there. But I have a doubt and possible a suggestion...

Why we need to always put the subproject within a folder? Why the `subproject` call cannot receive the path as argument? My suggestion is remove the need of a specific directory to subprojects. 

Thanks!

### PR DESCRIPTION
**Pull request by sarum9in**
Also added 2 links to subprojects page from reference manual.
Closes #2766

Assigning @jpakkane for review since it was his explanation and I believe it should be his decision whether we should include this verbatim.

### CODE REVIEW

File: docs/markdown/Subprojects.md [hunk 1]
```diff
@@ -77,3 +77,27 @@ subproject `b` and have `b` also use `a`.
 Meson ships with a dependency system to automatically obtain
 dependency subprojects. It is documented in the [Wrap dependency
 system manual](Wrap-dependency-system-manual.md).
+
+# Why all subproject should be inside a directory?
```
Comment (jon-turney) [inline docs/markdown/Subprojects.md]: I think the question being answered here is "Why must all subprojects be inside a single directory?"

Comment (sarum9in) [reply docs/markdown/Subprojects.md]: You are right, thank you.

**Comment by jpakkane** [comment]
Where did you get the text from? It feels like something I wrote but I can't tell where it's from.

**Comment by sarum9in** [comment]
I mentioned in the first comment "closes #2766". This is where you wrote it. And @nirbheek last commented that would be nice to have it in documentation, so I am sending this :)

**Pull request merged**